cmd/collectors/zapiperf/zapiperf.go

```diff
@@ -475,29 +505,68 @@ func (z *ZapiPerf) PollData() (map[string]*matrix.Matrix, error) {
 
 				// special case for workload_detail
 				if z.Query == objWorkloadDetail || z.Query == objWorkloadDetailVolume {
-					if name == "wait_time" || name == "service_time" {
-						if err := resourceLatency.AddValueString(instance, value); err != nil {
-							z.Logger.Error().
-								Stack().
-								Err(err).
-								Str("name", name).
-								Str("value", value).
-								Int("instIndex", instIndex).
-								Msg("Add resource-latency failed")
-						} else {
-							z.Logger.Trace().
-								Str("name", name).
-								Str("value", value).
-								Int("instIndex", instIndex).
-								Msg("Add resource-latency")
-							count++
+					for _, wm := range workloadDetailMetrics {
+						wMetric := curMat.GetMetric(layer + wm)
+
+						if wm == "resource_latency" && (name == "wait_time" || name == "service_time") {
+							if err := wMetric.AddValueString(instance, value); err != nil {
+								z.Logger.Error().
+									Stack().
+									Err(err).
+									Str("name", name).
+									Str("value", value).
+									Int("instIndex", instIndex).
+									Msg("Add resource_latency failed")
+							} else {
+								z.Logger.Trace().
+									Str("name", name).
+									Str("value", value).
+									Int("instIndex", instIndex).
+									Msg("Add resource_latency")
+								count++
+							}
+							continue
```

Comment:
maybe add `continue` for all blocks or none of the blocks? Doesn't look like any of the blocks need it, although it might be better to move `if name == "visits" {
continue
}` to the top